crates/astria-composer/src/executor/builder.rs

```diff
@@ -49,12 +47,17 @@ impl Builder {
         let sequencer_client = sequencer_client::HttpClient::new(sequencer_url.as_str())
             .wrap_err("failed constructing sequencer client")?;
         let (status, _) = watch::channel(Status::new());
-        let mut private_key_bytes: [u8; 32] = hex::decode(private_key.expose_secret())
-            .wrap_err("failed to decode private key bytes from hex string")?
+        let private_key_hex = fs::read_to_string(&private_key_file)
+            .wrap_err_with(|| format!("failed to read private key at `{private_key_file}`"))?;
+        let private_key_bytes: [u8; 32] = hex::decode(private_key_hex.trim())
+            .wrap_err_with(|| {
+                format!("failed to hex-decode private key bytes in `{private_key_file}`")
+            })?
             .try_into()
-            .map_err(|_| eyre!("invalid private key length; must be 32 bytes"))?;
+            .map_err(|_| {
+                eyre!("invalid private key length in `{private_key_file}`; must be 32 bytes")
+            })?;
         let sequencer_key = SigningKey::from(private_key_bytes);
```

Comment:
Maybe tuck this in a function so we can have single `wrap_err_with`?

```rust
fn read_signing_key_from_file<P: AsRef<Path>>(path: P) -> eyre::Result<SingingKey> {
    todo!()
}

read_signing_key_from_file(&private_key_file)
    .wrap_err_with(|| format!("failed reading signing key from file at path `{private_key_file}`)?
```

Reply:
Done in [840706c](https://github.com/astriaorg/astria/pull/1074/commits/840706c736bcaeb33cb74baf594fa6bc3429ec8b).